src/kili/queries/asset/helpers.py

```diff
@@ -42,26 +43,59 @@ def get_file_extension_from_headers(url) -> Optional[str]:
             with requests.get(url, timeout=20) as response:
                 response.raise_for_status()
                 headers = response.headers
-        content_type = headers["content-type"]
-        return guess_extension(content_type)
-
-
-def get_download_path(url: str, external_id: str, local_dir_path: Path) -> Path:
+        if "content-type" in headers:
+            content_type = headers["content-type"]
+            return guess_extension(content_type)
+    return None
+
+
+def get_file_name_from_url(url) -> str:
+    """guess the filename of a file url"""
+    path = urlparse(url).path
+
+    # video frames
+    if "/api/label/v2/files" in path and "/frame/" in url:
```

Comment:
This is true for frames hosted on Kili, i.e an asset imported from the UI. However, we can provide frames hosted elsewhere and they will have a different url format

Reply:
I think that we have to stick with the externalId to infer the filename. We can provide an index as argument for frames to create a name like externalId/0001.png, externalId/0002.png etc
We do it in the UI for the Yolo export, you can have a look in the service label-export in the kili codebase.

Reply:
I tested it, and the format used for the tree and names is:
Images/external_id1_0001.ext
Images/external_id1_0002.ext
...
Images/external_id2_0001.ext
Images/external_id2_0002.ext
...
...

So the tree is different that the choice that you take. We have to discuss tomorrow if it is smart to change that tree format in the assets queries compared to the export feature.

Reply:
Better stay consistent with the UI for now